cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ArtifactHttpHandler.java

```diff
@@ -688,6 +688,8 @@ public void deleteArtifact(HttpRequest request, HttpResponder responder,
     try {
       artifactRepository.deleteArtifact(artifactId);
       responder.sendStatus(HttpResponseStatus.OK);
+    } catch (ArtifactNotFoundException e) {
+      responder.sendString(HttpResponseStatus.NOT_FOUND, "Artifact " + artifactId + " not found.");
```

Comment:
Basically all occurrences of catching `ArtifactNotFoundException` and sending 404 should be removed from that class.